lib/openhab/core/actions/http.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module OpenHAB
+  module Core
+    module Actions
+      # @see https://www.openhab.org/docs/configuration/actions.html#http-actions HTTP Actions
+      class HTTP
+        class << self
+          #
+          # Sends an HTTP GET request and returns the result as a String.
+          #
+          # @param [String] url
+          # @param [Hash<String, String>] headers
+          # @param [Duration, int, nil] timeout Timeout (in milliseconds, if given as an Integer)
+          # @return [String] the response body
+          # @return [nil] if an error occurred
+          #
+          def send_http_get_request(url, headers: {}, timeout: nil)
+            timeout ||= 5_000
+            timeout = (timeout.to_f * 1_000).to_i if timeout.is_a?(Duration)
```

Comment:
Ooops sorry.... maybe `timeout.to_millis` ?

Reply:
good call. I was looking through https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html and not seeing it. but maybe that's cause I saw getNano and getSeconds